src/Domain/LeanCode.DomainModels/Ulids/Ulid.cs

```diff
@@ -0,0 +1,884 @@
+        30,
+        31,
+        255,
+        255,
+        255,
+        255,
+        255,
+        255,
+        10,
+        11,
+        12,
+        13,
+        14,
+        15,
+        16,
+        17,
+        255,
+        18,
+        19,
+        255,
+        20,
+        21,
+        255,
+        22,
+        23,
+        24,
+        25,
+        26,
+        255,
+        27,
+        28,
+        29,
+        30,
+        31
+    };
+
+    private static readonly DateTimeOffset UnixEpoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+    public static readonly Ulid MinValue = new Ulid(
+        UnixEpoch.ToUnixTimeMilliseconds(),
+        new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }
+    );
+
+    public static readonly Ulid MaxValue = new Ulid(
+        DateTimeOffset.MaxValue.ToUnixTimeMilliseconds(),
+        new byte[] { 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 }
+    );
+
+    public static readonly Ulid Empty;
+
+    // Core
+
+    // Timestamp(48bits)
+    [FieldOffset(0)]
+    private readonly byte timestamp0;
+
+    [FieldOffset(1)]
+    private readonly byte timestamp1;
+
+    [FieldOffset(2)]
+    private readonly byte timestamp2;
+
+    [FieldOffset(3)]
+    private readonly byte timestamp3;
+
+    [FieldOffset(4)]
+    private readonly byte timestamp4;
+
+    [FieldOffset(5)]
+    private readonly byte timestamp5;
+
+    // Randomness(80bits)
+    [FieldOffset(6)]
+    private readonly byte randomness0;
+
+    [FieldOffset(7)]
+    private readonly byte randomness1;
+
+    [FieldOffset(8)]
+    private readonly byte randomness2;
+
+    [FieldOffset(9)]
+    private readonly byte randomness3;
+
+    [FieldOffset(10)]
+    private readonly byte randomness4;
+
+    [FieldOffset(11)]
+    private readonly byte randomness5;
+
+    [FieldOffset(12)]
+    private readonly byte randomness6;
+
+    [FieldOffset(13)]
+    private readonly byte randomness7;
+
+    [FieldOffset(14)]
+    private readonly byte randomness8;
+
+    [FieldOffset(15)]
+    private readonly byte randomness9;
+
+    [IgnoreDataMember]
+    [SuppressMessage("?", "CA1819", Justification = "Fresh array is allocated per call")]
+    public byte[] Random =>
+        new byte[]
+        {
+            randomness0,
+            randomness1,
+            randomness2,
+            randomness3,
+            randomness4,
+            randomness5,
+            randomness6,
+            randomness7,
+            randomness8,
+            randomness9,
+        };
+
+    [IgnoreDataMember]
+    public DateTimeOffset Time
+    {
+        get
+        {
+            Span<byte> buffer = stackalloc byte[8];
+            buffer[0] = timestamp5;
+            buffer[1] = timestamp4;
+            buffer[2] = timestamp3;
+            buffer[3] = timestamp2;
+            buffer[4] = timestamp1;
+            buffer[5] = timestamp0; // [6], [7] = 0
+
+            var timestampMilliseconds = Unsafe.As<byte, long>(ref MemoryMarshal.GetReference(buffer));
```

Comment:
What about endianness here? Wouldn't it be safer to use something like `BinaryPrimitives.ReadInt64LittleEndian` instead?